modules/publicize/publicize-jetpack.php

```diff
@@ -231,20 +238,35 @@ function display_disconnected() {
 
 	function globalization() {
 		if ( 'on' == $_REQUEST['global'] ) {
-			$id = $_REQUEST['connection'];
+			$globalize_connection = $_REQUEST['connection'];
 
 			if ( ! current_user_can( $this->GLOBAL_CAP ) ) {
 				return;
 			}
 
-			Jetpack::load_xml_rpc_client();
-			$xml = new Jetpack_IXR_Client();
-			$xml->query( 'jetpack.globalizePublicizeConnection', $id, 'globalize' );
+			$this->globalize_connection( $connection_id );
+		}
+	}
+
+	function globalize_connection( $connection_id ) {
+		Jetpack::load_xml_rpc_client();
+		$xml = new Jetpack_IXR_Client();
+		$xml->query( 'jetpack.globalizePublicizeConnection', $connection_id, 'globalize' );
 
-			if ( ! $xml->isError() ) {
-				$response = $xml->getResponse();
-				Jetpack_Options::update_option( 'publicize_connections', $response );
-			}
+		if ( ! $xml->isError() ) {
+			$response = $xml->getResponse();
+			$this->receive_updated_publicize_connections( $response );
+		}
+	}
+
+	function unglobalize_connection( $connection_id ) {
+		Jetpack::load_xml_rpc_client();
+		$xml = new Jetpack_IXR_Client();
+		$xml->query( 'jetpack.globalizePublicizeConnection', $connection_id, 'unglobalize' );
+
+		if ( ! $xml->isError() ) {
+			$response = $xml->getResponse();
+			$this->receive_updated_publicize_connections( $response );
```

Comment:
Since both methods have almost identical implementations, can we merge them and make the `'globalize'`/`'unglobalize'` param to `$xml->query` an argument?

Reply:
We could - despite the number of red and green lines, I was going for minimal changes :) If you feel strongly about it, let's get it in. Otherwise, I think it can be a different PR.